src/monitord/monitord.c

```diff
@@ -8,129 +8,89 @@
  * Foundation
  */
 
+#include "cJSON.h"
+#include "debug_op.h"
+#include "hash_op.h"
 #include "os_err.h"
 #include "shared.h"
 #include "monitord.h"
 #include "config/config.h"
+#include "string_op.h"
+#include "wazuh_db/wdb.h"
+#include "time.h"
 
 /* Global variables */
 monitor_config mond;
+OSHash* agents_to_alert_hash;
+monitor_time_control mond_time_control;
 
 void Monitord()
 {
-    time_t tm;
-    int counter = 0;
-    struct tm tm_result = { .tm_sec = 0 };
-
     char path[PATH_MAX];
     char path_json[PATH_MAX];
-    struct stat buf;
-    off_t size;
-
-    int today = 0;
-    int thismonth = 0;
-    int thisyear = 0;
-
-    char str[OS_SIZE_1024 + 1];
 
     /* Wait a few seconds to settle */
     sleep(10);
 
-    memset(str, '\0', OS_SIZE_1024 + 1);
-
-    /* Get current time before starting */
-    tm = time(NULL);
-    localtime_r(&tm, &tm_result);
-
-    today = tm_result.tm_mday;
-    thismonth = tm_result.tm_mon;
-    thisyear = tm_result.tm_year + 1900;
-
     /* Set internal log path to rotate them */
 #ifdef WIN32
-    // ossec.log
+    /* ossec.log */
     snprintf(path, PATH_MAX, "%s", LOGFILE);
-    // ossec.json
+    /* ossec.json */
     snprintf(path_json, PATH_MAX, "%s", LOGJSONFILE);
 #else
-    // /var/ossec/logs/ossec.log
+    /* /var/ossec/logs/ossec.log */
     snprintf(path, PATH_MAX, "%s%s", isChroot() ? "" : DEFAULTDIR, LOGFILE);
-    // /var/ossec/logs/ossec.json
+    /* /var/ossec/logs/ossec.json */
     snprintf(path_json, PATH_MAX, "%s%s", isChroot() ? "" : DEFAULTDIR, LOGJSONFILE);
 #endif
 
-    /* Connect to the message queue or exit */
-    if ((mond.a_queue = StartMQ(DEFAULTQUEUE, WRITE, INFINITE_OPENQ_ATTEMPTS)) < 0) {
-        merror_exit(QUEUE_FATAL, DEFAULTQUEUE);
-    }
-
-    /* Send startup message */
-    snprintf(str, OS_SIZE_1024 - 1, OS_AD_STARTED);
-    if (SendMSG(mond.a_queue, str, ARGV0,
-                LOCALFILE_MQ) < 0) {
-        merror(QUEUE_SEND);
-    }
+    /* Connect to the message queue, infinite attempts */
+    monitor_queue_connect();
```

Comment:
StartMQ with INFINITE_OPENQ_ATTEMPTS is not truly "infinit" There are errors that will make the method return, like an eror onOS_BindUnixDomain().
Here, we are modifing the initialization because an error like this will not generate an exit of the process.
Is this correct?